kcqrs-core/src/main/kotlin/com/clouway/kcqrs/core/CommonValidators.kt

```diff
@@ -0,0 +1,48 @@
+package com.clouway.kcqrs.core
+
+private val emailRegex = kotlin.text.Regex("^[_A-Za-z0-9-\\+]+(\\.[_A-Za-z0-9-]+)*@[A-Za-z0-9-]+(\\.[A-Za-z0-9]+)*(\\.[A-Za-z]{2,})$")
+
+/**
+ * @author Miroslav Genov (miroslav.genov@clouway.com)
+ */
+val anEmailAddress = object : Validator<String> {
+    override fun validate(fieldValue: String): Boolean {
+        return fieldValue.matches(emailRegex)
+    }
+}
+
+fun <T> equal(expected: T?): Validator<T?> {
+    return object : Validator<T?> {
+        override fun validate(fieldValue: T?): Boolean {
+            return expected == fieldValue
+        }
+    }
+}
+
+fun <T> notEqualTo(expected: T?): Validator<T?> {
+    return object : Validator<T?> {
+        override fun validate(fieldValue: T?): Boolean {
+            return expected != fieldValue
+        }
+    }
+}
+
+fun <T : Comparable<T>> greatherThan(expected: T): Validator<T> {
+    return object : Validator<T> {
+        override fun validate(fieldValue: T): Boolean {
+            return fieldValue > expected
+        }
+    }
+}
+
+fun <T : Comparable<T>> lowerThan(expected: T): Validator<T> {
+    return object : Validator<T> {
+        override fun validate(fieldValue: T): Boolean {
+            return fieldValue < expected
+        }
+    }
+}
+
+infix fun <T> T.shouldBe(validator: Validator<T>): Boolean {
```

Comment:
It is strange for me when you call ``` shouldBe(validator) ``` and you pass validator to this method.
I think it will be better if we call ``` shouldMatch(validator) ```.

Reply:
Can you show me an example usage ?

Reply:
with shouldMatch I mean.

Reply:
No, I cant. It`s just sound better to me as a name.

Reply:
You looking in wrong direction, the right one is how validation is defined and not the signature of this function. Also keep in mind that it's an infix, e.g belongs to type T.

Reply:
For example:
```kotlin
createOn shouldBe inRange(LocalDateTime.now(), LocalDateTime.MAX)
```
vs
```
createOn shouldMatch inRange(LocalDateTime.now(), LocalDateTime.MAX)
createOn shouldMatch rangeValues(LocalDateTime.now(), LocalDateTime.MAX)
createOn shouldBe inRange(LocalDateTime.now(), LocalDateTime.MAX)
```

Reply:
I see now your point of view.